### ISSUE TITLE
GKE single-line JSON logging

### ISSUE THREAD

**Issue by maresja1** [kubernetes/kubernetes#72460]
**What happened**:

When upgrading from GKE 1.11.3 to 1.11.4 single-line json logging stops working.

For a line logged to stdout:

```json
{"severity":"NOTICE","logger":"org.springframework.web.socket.config.WebSocketMessageBrokerStats","thread":"MessageBroker-1","serviceContext":{"service":"product-container","version":"1.8.2"},"message":"WebSocketSession[1 current WS(1)-HttpStream(0)-HttpPoll(0), 1739 total, 0 closed abnormally (0 connect failure, 0 send limit, 0 transport error)], stompSubProtocol[processed CONNECT(1729)-CONNECTED(1729)-DISCONNECT(318)], stompBrokerRelay[2 sessions, ReactorNettyTcpClient[reactor.netty.tcp.TcpClientDoOn@89ff043] (available), processed CONNECT(1730)-CONNECTED(1731)-DISCONNECT(1695)], inboundChannel[pool size \u003d 2, active threads \u003d 0, queued tasks \u003d 0, completed tasks \u003d 152205], outboundChannelpool size \u003d 2, active threads \u003d 0, queued tasks \u003d 0, completed tasks \u003d 59060], sockJsScheduler[pool size \u003d 1, active threads \u003d 1, queued tasks \u003d 1, completed tasks \u003d 170270]","logging.googleapis.com/sourceLocation":{"line":124,"function":"org.springframework.web.socket.config.WebSocketMessageBrokerStats.lambda$initLoggingTask$0","file":"org/springframework/web/socket/config/WebSocketMessageBrokerStats.java"},"timestamp":{"nanos":130000000,"seconds":1546200086}}
```

On the GKE versions up to 1.11.3, there is a jsonPayload in Stackdriver logging corresponding to the structure of the JSON and severity is taken into account. On the versions of 1.11.4 and above, there is only textPayload containing unparsed json:

![image](https://user-images.githubusercontent.com/1552179/50567367-67af7500-0d44-11e9-9e1c-9ff533736122.png) 

**What you expected to happen**:

I expect to get jsonPayload with structured data as I got on the version 1.11.3:

![image](https://user-images.githubusercontent.com/1552179/50567393-b0ffc480-0d44-11e9-9e5c-a9b48ce7e9b0.png)

**How to reproduce it (as minimally and precisely as possible)**:

```bash
$ kubectl run --image=busybox log-test -- echo '{"message":"asd","severity":"WARNING"}'
```

***Workaround***

Downgrade and stay on 1.11.3-gke.X (at the moment 1.11.3-gke.24)

**Environment**:
- Kubernetes version (use `kubectl version`): 1.11.4-gke.X and 1.11.5-gke.X
- Cloud provider or hardware configuration: GKE (GCP)
- OS (e.g. from /etc/os-release): GKE standard image
- Others:

This did not manifest right after upgrade to version 1.11.5, but on 18.12.2018, (the upgrade was on 9.12.2018). It seems to be related with upgrade from fluentd-gcp-v3.1.0 to fluentd-gcp-v3.1.1 that resulted in upgrade of its sub-components.

For 18.12.2018 I see in activity log:

google.container.internal.ClusterManagerInternal.UpdateClusterInternal was executed

as I have automatic updates enabled, I guess it is a result of that.

/kind bug

**Comment by maxekman** [comment]
I have also experienced this same issue, started Dec 21 after upgrading the cluster to v1.11.4.

It looks like it could be solved in v0.7.3 of the Google Cloud plugin: https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/releases/tag/v0.7.3

More specifically I suspect it happens because this check fails: https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/v0.7.3/lib/fluent/plugin/out_google_cloud.rb#L626

**Comment by sebglon** [comment]
the same for me on 1.11.6-gke.0

**Comment by lborie** [comment]
The same on 1.11.5-gke.5.
A bit confused by this problem and the impact around it

**Comment by mradzevicius** [comment]
Any steps to get this working again?

**Comment by billjh** [comment]
The same on 1.11.5-gke.5

**Comment by sebglon** [comment]
solved for me on 1.11.6-gke.2

**Comment by billjh** [comment]
Solved after upgrade to 1.11.6-gke.3
